### PR TITLE
feat!: use `x/governors` instead of `x/staking`

### PR DESCRIPTION
**Pull request by mtsitrin**
## Description

----

Closes #XXX

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow-up issues.

PR review checkboxes:

I have...

- [ ]  Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ]  Targeted PR against the correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to the GitHub issue with discussion and accepted design
- [ ]  Targets only one GitHub issue
- [ ]  Wrote unit and integration tests
- [ ]  Wrote relevant migration scripts if necessary
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)
- [ ]  Updated the scripts for local run, e.g genesis_config_commands.sh if the PR changes parameters
- [ ]  Add an issue in the [e2e-tests repo](https://github.com/dymensionxyz/e2e-tests) if necessary

SDK Checklist
- [ ] Import/Export Genesis
- [ ] Registered Invariants
- [ ] Registered Events
- [ ] Updated openapi.yaml
- [ ] No usage of go `map`
- [ ] No usage of `time.Now()`
- [ ] Used fixed point arithmetic and not float arithmetic
- [ ] Avoid panicking in Begin/End block as much as possible
- [ ] No unexpected math Overflow
- [ ] Used `sendCoin` and not `SendCoins`
- [ ] Out-of-block compute is bounded
- [ ] No serialized ID at the end of store keys
- [ ] UInt to byte conversion should use BigEndian

Full security checklist [here](https://www.faulttolerant.xyz/2024-01-16-cosmos-security-1/)

----;

For Reviewer:

- [ ]  Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  Confirmed all author checklist items have been addressed

---;

After reviewer approval:

- [ ]  In case the PR targets the main branch, PR should not be squash merge in order to keep meaningful git history.
- [ ]  In case the PR targets a release branch, PR must be rebased.

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -56,8 +56,9 @@ import (
 	"github.com/cosmos/cosmos-sdk/x/feegrant"
 	feegrantkeeper "github.com/cosmos/cosmos-sdk/x/feegrant/keeper"
 	feegrantmodule "github.com/cosmos/cosmos-sdk/x/feegrant/module"
-	"github.com/cosmos/cosmos-sdk/x/genutil"
-	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+
+	// "github.com/cosmos/cosmos-sdk/x/genutil"
+	// genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
```
Comment (danwt) [inline app/app.go:61]: can we just delete instead of commenting
or if commenting out, needs a TODO: and an explanation

File: app/app.go [hunk 2]
```diff
@@ -194,10 +195,10 @@ var (
 	ModuleBasics = module.NewBasicManager(
 		auth.AppModuleBasic{},
 		authzmodule.AppModuleBasic{},
-		genutil.AppModuleBasic{},
+		// genutil.AppModuleBasic{},
```
Comment (danwt) [inline app/app.go:198]: ditto re commenting

File: app/app.go [hunk 3]
```diff
@@ -575,10 +576,10 @@ func NewRollapp(
...
 
 	modules := []module.AppModule{
-		genutil.NewAppModule(
-			app.AccountKeeper, app.StakingKeeper, app.BaseApp.DeliverTx,
-			encodingConfig.TxConfig,
-		),
+		// genutil.NewAppModule(
+		// 	app.AccountKeeper, app.StakingKeeper, app.BaseApp.DeliverTx,
+		// 	encodingConfig.TxConfig,
+		// ),
```
Comment (danwt) [inline app/app.go:582]: ditto re commenting

File: app/export.go [hunk 4]
```diff
@@ -33,13 +32,11 @@ func (app *App) ExportAppStateAndValidators(
 		return servertypes.ExportedApp{}, err
 	}
 
-	validators, err := staking.WriteValidators(ctx, app.StakingKeeper.Keeper)
-	if err != nil {
-		return servertypes.ExportedApp{}, err
-	}
+	// TODO: read validators from sequencer modules
+
```
Comment (danwt) [inline app/export.go:36]: why dont we do it in this PR?

**Review by danwt — DISMISSED** [review]
approved but not a fan of commenting out code, at all
please just delete instead, and leave a comment with a link to a github commit containing the commments

Or write todos

**Comment by omritoptix** [comment]
@mtsitrin e2e tests fails. possibly due to genesis changes.

**Comment by omritoptix** [comment]
@mtsitrin this change breaks the relayer. when trying to create an ibc connection, getting the following error (while trying to query for the unbonding period):

`Error: error creating clients: failed to create client on dst chain{dymension_100-1}: failed to get trusting period for chain{rollappevm_9988-1}: failed to query unbonding period from ccvconsumer, staking & fallback : failed to make ccvconsumer params request: rpc error: code = Unknown desc = ccvconsumer: unknown subspace: unknown request: failed to make staking params request: rpc error: code = Unknown desc = staking: unknown subspace: unknown request : rpc error: code = Unknown desc = unknown query path: unknown request`

Screenshot 2024-04-26 at 0 33 02

**Comment by omritoptix** [comment]
closing for now until we decided the direction with that

**Pull request closed without merging**